sphinx/ext/intersphinx.py

```diff
@@ -570,34 +574,49 @@ def get_inventory_and_name_suffix(self, name: str) -> tuple[str | None, str]:
             raise ValueError(msg)
 
     def get_role_name(self, name: str) -> tuple[str, str] | None:
+        """Given a string like ``domain:name`` or ``name``,
+        return a tuple of the (domain, role name),
+        or None if no available domain/role can be found.
+
+        The ``name`` can be either a role name or an object name.
+        """
         names = name.split(':')
         if len(names) == 1:
-            # role
             default_domain = self.env.temp_data.get('default_domain')
             domain = default_domain.name if default_domain else None
-            role = names[0]
+            name = names[0]
         elif len(names) == 2:
-            # domain:role:
             domain = names[0]
-            role = names[1]
+            name = names[1]
         else:
             return None
 
-        if domain and self.is_existent_role(domain, role):
+        if domain and (role := self.is_existent_role(domain, name)):
             return (domain, role)
-        elif self.is_existent_role('std', role):
+        elif (role := self.is_existent_role('std', name)):
```

Comment:
Maybe say that the standard domain is used as a fallback when we have `:name` or when we have `domain:name` but no role can be found.

Reply:
done